cpp/dolfinx/common/constexpr_type.h

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2025 Paul T. Kühner
+//
+// This file is part of DOLFINx (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+
+#include <type_traits>
+
+namespace dolfinx::common
+{
+/// @private Concept defining a variadic compile time or runtime variable. T
+/// indicates the type that is stored and V the value. Either V equals T, i.e.
+/// it is a runtime variable or V defines a compile time value V::value of type
+/// T.
+/// @tparam T type of the value to be stored.
+/// @tparam V container type. Usually T for a runtime variable or a
+/// std::integral_constant<T, ...> for a compile time constant.
+template <typename T, typename V>
+concept ConstexprType = std::is_same_v<T, V> || (requires {
+                          std::is_fundamental_v<T>;
```

Comment:
I think there should be a `requires` here - without requires,  isn't it just an expression?